src/Utils/ASTDefinitionBuilder.php

```diff
@@ -188,11 +188,17 @@ private function buildWrappedType(TypeNode $typeNode): Type
      */
     public function buildType($ref): Type
     {
-        if (is_string($ref)) {
-            return $this->internalBuildType($ref);
+        $type = null;
+
+        if ($ref instanceof TypeDefinitionNode) {
+            $type = $this->internalBuildType($ref->getName()->value, $ref);
+        } elseif ($ref instanceof NamedTypeNode) {
+            $type = $this->internalBuildType($ref->name->value, $ref);
```

Comment:
Why don't we add `getName()` to that as well and go back to the simple implementation?

Reply:
Properties are fine for classes 🤷‍♂️